core/Azure.Mcp.Core/src/Services/Azure/Tenant/TenantService.cs

```diff
@@ -40,42 +52,54 @@ public async Task<List<TenantResource>> GetTenants()
         return results;
     }
 
-    public bool IsTenantId(string tenant)
+    /// <inheritdoc/>
+    public bool IsTenantId(string tenantId)
     {
-        return Guid.TryParse(tenant, out _);
+        return Guid.TryParse(tenantId, out _);
     }
 
-    public async Task<string?> GetTenantId(string tenant)
+    /// <inheritdoc/>
+    public async Task<string> GetTenantId(string tenantIdOrName)
     {
-        if (IsTenantId(tenant))
+        if (IsTenantId(tenantIdOrName))
         {
-            return tenant;
+            return tenantIdOrName;
         }
 
-        return await GetTenantIdByName(tenant);
+        return await GetTenantIdByName(tenantIdOrName);
     }
 
-    public async Task<string?> GetTenantIdByName(string tenantName)
+    /// <inheritdoc/>
+    public async Task<string> GetTenantIdByName(string tenantName)
     {
         var tenants = await GetTenants();
         var tenant = tenants.FirstOrDefault(t => t.Data.DisplayName?.Equals(tenantName, StringComparison.OrdinalIgnoreCase) == true) ??
             throw new Exception($"Could not find tenant with name {tenantName}");
 
-        if (tenant.Data.TenantId == null)
+        string? tenantId = tenant.Data.TenantId?.ToString();
+        if (tenantId == null)
             throw new InvalidOperationException($"Tenant {tenantName} has a null TenantId");
 
-        return tenant.Data.TenantId.ToString();
+        return tenantId.ToString();
     }
 
-    public async Task<string?> GetTenantNameById(string tenantId)
+    /// <inheritdoc/>
+    public async Task<string> GetTenantNameById(string tenantId)
     {
         var tenants = await GetTenants();
         var tenant = tenants.FirstOrDefault(t => t.Data.TenantId?.ToString().Equals(tenantId, StringComparison.OrdinalIgnoreCase) == true) ??
             throw new Exception($"Could not find tenant with ID {tenantId}");
 
-        if (tenant.Data.DisplayName == null)
+        string? tenantName = tenant.Data.DisplayName;
+        if (tenantName == null)
             throw new InvalidOperationException($"Tenant with ID {tenantId} has a null DisplayName");
 
-        return tenant.Data.DisplayName;
+        return tenantName;
+    }
+
+    /// <inheritdoc/>
+    public async Task<TokenCredential> GetTokenAsync(string? tenantId, CancellationToken cancellationToken)
+    {
+        return await _credentialProvider.GetTokenAsync(tenantId, cancellationToken);
```

Comment:
Nit: one minor thing I got confused about - the method name `GetTokenAsync` initially made me think it returns an `Azure.Core.AccessToken`, but it actually returns a `Azure.Core.TokenCredential`. Since `Azure.Core.TokenCredential` itself has a `GetTokenAsync` method, should we consider renaming this to `ITenantService::GetTokenCredentialAsync` and `IAzureTokenCredentialProvider::GetTokenCredentialAsync` for better clarity?

Reply:
Sure, I can do that.

Reply:
Updated locally. Pending push.